testpath/asserts.py

```diff
@@ -137,3 +139,55 @@ def assert_not_islink(path, msg=None):
         if msg is None:
             msg = "Path is a symlink: %r" % path
         raise AssertionError(msg)
+
+def assert_ispipe(path, follow_symlinks=True, msg=None):
+    """Assert that path exists and is a named pipe (FIFO).
+
+    With follow_symlinks=True, the default, this will pass if path is a symlink
+    to a regular file. With follow_symlinks=False, it will fail in that case.
+    """
+    path = _strpath(path)
+    st = _stat_for_assert(path, follow_symlinks, msg)
+    if not stat.S_ISFIFO(st.st_mode):
+        if msg is None:
+            msg = "Path exists, but is not a named pipe: %r" % path
+        raise AssertionError(msg)
+
+def assert_not_ispipe(path, follow_symlinks=True, msg=None):
+    """Assert that path exists but is not a named pipe (FIFO).
+
+    With follow_symlinks=True, the default, this will fail if path is a symlink
+    to a regular file. With follow_symlinks=False, it will pass in that case.
+    """
+    path = _strpath(path)
+    st = _stat_for_assert(path, follow_symlinks, msg)
+    if stat.S_ISFIFO(st.st_mode):
+        if msg is None:
+            msg = "Path is a named pipe: %r" % path
+        raise AssertionError(msg)
+
+def assert_issocket(path, follow_symlinks=True, msg=None):
+    """Assert that path exists and is a named pipe (FIFO).
```

Comment:
These docstrings should probably say 'Unix domain socket' rather than 'named pipe'.